lib/ui/SpinniesManager.ts

```diff
@@ -9,32 +8,60 @@ Permission is hereby granted, free of charge, to any person obtaining a copy of
 The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 **/
 
-const readline = require('readline');
-const chalk = require('chalk');
-const cliCursor = require('cli-cursor');
-const {
+import readline from 'readline';
+import chalk from 'chalk';
+import cliCursor from 'cli-cursor';
+import {
   breakText,
   cleanStream,
   colorOptions,
   getLinesLength,
   purgeSpinnerOptions,
   purgeSpinnersOptions,
+  Spinner,
   SPINNERS,
   terminalSupportsUnicode,
   writeStream,
-} = require('./spinniesUtils');
+  SpinnerOptions as BaseSpinnerOptions,
+  VALID_STATUSES,
+  prefixOptions,
+} from './spinniesUtils';
+
+interface SpinnerState extends BaseSpinnerOptions {
+  name?: string;
+}
+
+function safeColor(text: string, color?: string): string {
+  const chalkFn = chalk[color as keyof typeof chalk];
+
+  if (typeof chalkFn === 'function') {
+    return (chalkFn as (text: string) => string)(text);
+  }
+  return text;
+}
 
 class SpinniesManager {
+  private options!: SpinnerState;
+  private spinners: Record<string, SpinnerState> = {};
+  private isCursorHidden: boolean = false;
+  private currentInterval: NodeJS.Timeout | null = null;
+  private stream: NodeJS.WriteStream = process.stderr;
+  private lineCount: number = 0;
+  private currentFrameIndex: number = 0;
+  private spin: boolean = false;
+
   constructor() {
     this.resetState();
   }
 
-  init(options = {}) {
+  init(options: Partial<SpinnerState> = {}): void {
     this.options = {
       spinnerColor: 'greenBright',
       succeedColor: 'green',
       failColor: 'red',
-      spinner: terminalSupportsUnicode() ? SPINNERS.dots : SPINNERS.dashes,
+      spinner: (terminalSupportsUnicode()
+        ? SPINNERS.dots
+        : SPINNERS.dashes) as Spinner,
```

Comment:
I don't think you need to cast here

Reply:
https://github.com/HubSpot/hubspot-cli/pull/1233/commits/571b4f72d40f4aebe7286e7c47bf86500c3b835e